src/lib.js

```diff
@@ -3491,6 +3492,158 @@ async function getForge(userProfile) {
   return output;
 }
 
+function calculateFarmingWeight(userProfile) {
+  const output = {
+    weight: 0,
+    crops: {},
+    bonuses: {
+      level: {
+        level: 0,
+        weight: 0,
+      },
+    },
+  };
+
+  const crops = {
+    CACTUS: {
+      name: "Cactus",
+      weight: 169_389.33,
+    },
+    CARROT_ITEM: {
+      name: "Carrot",
+      weight: 300_000,
+    },
+    "INK_SACK:3": {
+      name: "Cocoa Bean",
+      weight: 257_214.64,
+    },
+    MELON: {
+      name: "Melon",
+      weight: 435_466.47,
+    },
+    NETHER_STALK: {
+      name: "Nether Wart",
+      weight: 250_000,
+    },
+    POTATO_ITEM: {
+      name: "Potato",
+      weight: 300_000,
+    },
+    PUMPKIN: {
+      name: "Pumpkin",
+      weight: 87_095.11,
+    },
+    SUGAR_CANE: {
+      name: "Sugar Cane",
+      weight: 200_000,
+    },
+    WHEAT: {
+      name: "Wheat",
+      weight: 100_000,
+    },
+  };
+
+  if (userProfile?.collections) {
+    let weight = 0;
+    for (const [name, crop] of Object.entries(crops)) {
+      let total = userProfile.collections[name].amount;
```

Comment:
```suggestion
      let total = userProfile.collections[name]?.amount ?? 0;
```
Added a fallback to zero, as I'm not sure if locked collections would still appear in here.